PWGHF/HFC/Tasks/taskCorrelationD0Hadrons.cxx

```diff
@@ -249,11 +252,28 @@ struct HfTaskCorrelationD0Hadrons {
         continue;
       }
       //==============================================================================================================
-      if (signalStatus == ParticleTypeData::D0Only || (signalStatus == ParticleTypeData::D0D0barBoth)) {
-        registry.fill(HIST("hCorInfoWithCorrelationState"), deltaPhi, deltaEta, ptD, ptHadron, poolBin, massD, isAutoCorrelated, efficiencyWeight);
-      }
-      if (signalStatus == ParticleTypeData::D0barOnly || (signalStatus == ParticleTypeData::D0D0barBoth)) {
-        registry.fill(HIST("hCorInfoWithCorrelationState"), deltaPhi, deltaEta, ptD, ptHadron, poolBin, massDbar, isAutoCorrelated, efficiencyWeight);
+      if (isTowardTransverseAway) {
+        if (ptHadron < leadingParticlePtMin) {
+          continue;
+        }
+        if (signalStatus == ParticleTypeData::D0Only || (signalStatus == ParticleTypeData::D0D0barBoth)) {
+          if (-o2::constants::math::PI / 3. < deltaPhi && deltaPhi < o2::constants::math::PI / 3.) {
+            registry.fill(HIST("hToward"), massD, ptD, isAutoCorrelated, efficiencyWeight);
+          } else if (o2::constants::math::PI * 2. / 3. < deltaPhi && deltaPhi < o2::constants::math::PI * 4. / 3.) {
+            registry.fill(HIST("hAway"), massD, ptD, isAutoCorrelated, efficiencyWeight);
+          } else {
+            registry.fill(HIST("hTransverse"), massD, ptD, isAutoCorrelated, efficiencyWeight);
+          }
+        }
+        if (signalStatus == ParticleTypeData::D0barOnly || (signalStatus == ParticleTypeData::D0D0barBoth)) {
+          if (-o2::constants::math::PI / 3. < deltaPhi && deltaPhi < o2::constants::math::PI / 3.) {
+            registry.fill(HIST("hToward"), massD, ptD, isAutoCorrelated, efficiencyWeight);
+          } else if (o2::constants::math::PI * 2. / 3. < deltaPhi && deltaPhi < o2::constants::math::PI * 4. / 3.) {
+            registry.fill(HIST("hAway"), massD, ptD, isAutoCorrelated, efficiencyWeight);
+          } else {
+            registry.fill(HIST("hTransverse"), massD, ptD, isAutoCorrelated, efficiencyWeight);
+          }
```

Comment:
If you plan to add more code branched based on the region, I would suggest to define a function, which takes `deltaPhi` as argument and returns the region, which can be an `enum` constant.